packages/espressocash_app/lib/features/incoming_link_payments/services/ilp_service.dart

```diff
@@ -158,16 +169,64 @@ class ILPService implements Disposable {
 
     _refreshBalance();
 
+    _analytics.singleLinkReceived(amount: receiveAmount?.decimal);
+
     return payment.copyWith(
       status: ILPStatus.success(
         tx: status.tx,
+        receiveAmount: receiveAmount,
         fee: fee?.let(
           (fee) => CryptoAmount(value: fee, cryptoCurrency: Currency.usdc),
         ),
       ),
     );
   }
 
+  Future<CryptoAmount?> getUsdcAmount(String signature) async {
```

Comment:
It should be private. Also, it can throw an error, make sure that it's processed correctly.